### ISSUE TITLE
useInfiniteQueryParam generates invallid code when param does not exist.

### ISSUE THREAD

**Issue by bschoenmaeckers** [anymaniax/orval#393]
## What are the steps to reproduce this issue?

When useInfiniteQueryParam is enabled orval will generate invallid code for endpoints without the specified param.

In the orval config, useInfiniteQueryParam is set to `useInfiniteQueryParam: "page"`. 
And the openapi spec is as follows.

```yaml
openapi: 3.0.1
info: ...
paths:
  /path_with_query:
    get:
      parameters:
        - in: query
          name: name
          schema:
            type: string
      responses: ...
```

## What happens?

Orval will create a query function with 'page' query parameter but its type does not allow it.
```ts
const queryFn: QueryFunction<Awaited<ReturnType<typeof pathWithQuery>>> = ({ signal, pageParam }) => pathWithQuery({ page: pageParam, ...params }, { signal, ...axiosOptions });
```
This will error because params type does not contain the 'page' field
```ts
export type PathWithQueryParams = { name?: string }
```

## What were you expecting to happen?

Orval should only include the page param if it is defined in the api spec or include the useInfiniteQueryParam in every param type (e.g. PathWithQueryParams) automatically. 

## Any logs, error output, etc?

…

## Any other comments?

…

## What versions are you using?

**Package Version:** Orval v6.8.0

**Comment by RichardLindhout** [comment]
I have the same use case. I would like to skip infinite queries if page is not available

**Comment by JaneJeon** [comment]
Would definitely be interested in seeing this as well. I have some _portions_ of an API where the API structure is really dictated by some other admin framework, and trying to re-use it alongside the rest of the API when the parts dictated by the admin framework cannot support such infinite queries is unfortunately not possible.

The only solution I see is "splitting up" OpenAPI files for the same API endpoint (one for the parts I control, one for the parts I cannot), but it is a bit of a "dirty" solution and I'd definitely be interested in seeing an official solution to this.

**Comment by melloware** [comment]
@georgiev-anton another infinite to look at?

**Comment by melloware** [comment]
Can someone try 6.21.0 with this issue as infinite query updates have been made.

**Comment by georgiev-anton** [comment]
> Can someone try 6.21.0 with this issue as infinite query updates have been made.

yes, need to try with new changes

**Comment by koloboid** [comment]
Guys, I've tested 6.21.0, and have to say that issue is not fixed yet.

My case is the same as the topic starter - useXXXInfinite adds a missing param (in my case it's "offset") to all GET requests, which fails during typescript compilation.

I've made a temporary patch to Orval: https://github.com/koloboid/orval/commit/05a6370bdfdbd98277c3a7ff8a74fe6624839135

it works properly in my custom usage, but unfortunately, I have no time to make tests and clean up the code (I'm unsure it works well in all cases), so this patch is still in an inconsistent state. Hope it helps

**Comment by melloware** [comment]
Thank you for testing @koloboid

**Comment by madskchristensen** [comment]
What's the status of this? Currently experiencing the same issue. And what is a temporary workaround until it is fixed? 

My initial idea was to remove `useInfinite` and `useInfiniteQueryParam` from the global `output.override` in orval.config and specifying the setting for each operation where the endpoint's params supports my specified pageParam.

**Comment by destpat** [comment]
This issue is not fixed in version 6.24.0 as proposed by @madskchristensen. The current cleanest workaround is to deactivate `useInfinite` and `useInfiniteQueryParam` from the global `output.override`.

Example:


```
myapi: {
    input: {
      target: 'https://stoplight.io/api/v1/projects/my-api/api.json',
    },
    output: {
      override: {
        query: {
          useQuery: true,
          useSuspenseQuery: true,
        },
        operations: {
          'get-users': {
            query: {
              useQuery: true,
              useSuspenseQuery: true,
              useInfinite: true,
              useInfiniteQueryParam: 'page',
            },
          },
        },
      },
      target: './src',
      client: 'react-query',
      ...
    },
  },
```

In the example above `useInfinite` **hook** will be create only for `get-users` operation.

**Comment by kennyhei** [comment]
> koloboid@05a6370

Thank you, this fixed my issue as I have currently only one endpoint where I'd like to use `useInfinite` and `useInfiniteQueryParam` 🙏 Probably going to have more endpoints in the future so would be neat if this gets fixed.

**Comment by madskchristensen** [comment]
> This issue is not fixed in version 6.24.0 as proposed by @madskchristensen. The current cleanest workaround is to deactivate `useInfinite` and `useInfiniteQueryParam` from the global `output.override`.
> 
> Example:
> 
> ```
> myapi: {
>     input: {
>       target: 'https://stoplight.io/api/v1/projects/my-api/api.json',
>     },
>     output: {
>       override: {
>         query: {
>           useQuery: true,
>           useSuspenseQuery: true,
>         },
>         operations: {
>           'get-users': {
>             query: {
>               useQuery: true,
>               useSuspenseQuery: true,
>               useInfinite: true,
>               useInfiniteQueryParam: 'page',
>             },
>           },
>         },
>       },
>       target: './src',
>       client: 'react-query',
>       ...
>     },
>   },
> ```
> 
> In the example above `useInfinite` **hook** will be create only for `get-users` operation.

Can confirm this is also how I got around the issue for now. Would be very nice to see this fixed as infinite queries are very useful.

**Comment by melloware** [comment]
@madskchristensen i am still not sure if that fixes this issue or not. @lnkarma can you answer?
